torch/_dynamo/variables/torch.py

```diff
@@ -301,7 +302,7 @@ def call_function(
             assert len(args) == 2
             return VmapIncrementNestingCtxManagerVariable.create(
                 tx,
-                [guard_if_dyn(x) for x in args],
+                [unwrap_sym_or_const(x) for x in args],
```

Comment:
I don't think this is the right thing to call here. I suspect these should be proxies instead.

Reply:
@ezyang Do you know offhand how this should be held? Context is that this is what should be put in the Dynamo FX graph, and one of these should allow for a SymInt. At this point, some of these may be SymIntNode objects.

Reply:
The batch_size eventually gets passed to https://github.com/pytorch/pytorch/blob/63bbf712d889de8e07ff4f4f1896b3af3817b100/torch/_dynamo/variables/ctx_manager.py#L438-L443.

For that you want to be able to pass it a proxy and you also want to be able to retrieve the actual SymInt (for the lines above it that directly call `_vmap_increment_nesting`.

I would pass batch_size as the SymNodeVariable here (so that the VmapIncrementNestingCtxManagerVariable holds VariableTrackers) and then do then unwrapping inside it.

NB: there's likely a pre-existing bug here, in that we shouldn't actually be directly calling `torch._C._vmap_increment_nesting` (instead we should be simulating it in Dynamo). But I don't think this matters here